packages/store/src/Types.sol

```diff
@@ -1,84 +1,238 @@
 // SPDX-License-Identifier: MIT
 pragma solidity >=0.8.0;
 
-// TODO: add remaining types
+using { getStaticByteLength } for SchemaType global;
+
+// WARNING: SchemaType methods use hardcoded enum indexes, review them after any changes to the enum
+// TODO add and implement BYTES_ARRAY, STRING_ARRAY if they are needed (they are special nested arrays)
 enum SchemaType {
-  None, // The first `None` value ends the schema
-  Uint8,
-  Uint16,
-  Uint32,
-  Uint128,
-  Uint256,
-  Bytes4,
-  Bytes32,
-  Uint32Array,
-  Bytes24Array,
-  String,
-  Address,
-  AddressArray
+  UINT8,
```

Comment:
starting the enum here feels a little dangerous with no null/undefined? should we keep `None` or similar for index 0?

Reply:
I think the downside of `None` is it just adds extra complexity of checking that something isn't `None` in `Store`, and it complicates `getStaticByteLength` a tiny bit

Reply:
Since we store `numStaticFields` in the schema itself, we could live without `None` (without `numStaticFields` and no `None` schema type indexers couldn't distinguish between `none` and `uint8`, but with `numStaticFields` they can.)

Reply:
I don't think I have my head fully wrapped around the store stuff, so long as its easy to otherwise check if something is "unset" then we should be fine here.